src/ucioption.cpp

```diff
@@ -97,16 +97,20 @@ void on_variant_change(const Option &o) {
 
     const Variant* v = variants.find(o)->second;
 
+    const int typeCount = popcount(v->pieceTypes);
+    const int typeCount5bits = (typeCount > 15) ? 15 : typeCount;
+    const int typeCount9bits = typeCount - typeCount5bits;
     const int dataSize = (v->maxFile + 1) * (v->maxRank + 1) + v->nnueMaxPieces * 5
-                        + popcount(v->pieceTypes) * 2 * 5 + 50 > 512 ? 1024 : 512;
+                        + typeCount5bits * 2 * 5 + typeCount9bits * 2 * 9
```

Comment:
these are the bits for the pocket counts, see https://github.com/fairy-stockfish/variant-nnue-pytorch/wiki/Technical-details#training-data-format, the `v->nnueMaxPieces * 5` is the one for the pieces on the board. It could e.g. be `v->nnueMaxPieces * (typeCount > 15 ? 9 : 5)`. I think using worst case estimates here (all pieces are of the last type) is reasonable in order to be safe.

Reply:
Sorry, I should have picked that one up, I broke a long-standing rule to not code just before bed! 😴

Reply:
thanks. The other change should be reverted though, since the 5 bit for the pockets are counts per type (5 bit = [0, 31]), i.e., they are fixed size and not related to the huffman encoding. However, it is using 7 bit when using the bigger data size, see https://github.com/fairy-stockfish/variant-nnue-tools/blob/cc4094b8c6e03e82435cc5ec8dd38e3c2088173c/src/tools/sfen_packer.cpp#L200-L202
but that does not need to be considered here. It could just be added as an additional criterion, but this isn't really related to this PR, so no need to add it here unless you would like to.

Reply:
Sorry if there might have been a misunderstanding. The changes regarding the bits for the pieces in hand still needs to be reverted. The comments on this formula I will add once this PR is merged to avoid merge conflicts.

Reply:
Hi, there was a misunderstanding it seems, but I have also been busy preparing to move house so haven't gotten around to fixing this yet. I'll make it a draft again for the moment until I have some time to revert the pieces in hand.

Reply:
Ah, ok. There is no rush of course, so take your time. Or if you want I could also push the changes to the PR from my side.